openstack/networking/v2/extensions/bgpvpns/requests.go

```diff
@@ -0,0 +1,472 @@
+	return
+}
+
+// CreateOptsBuilder allows extensions to add additional parameters to the
+// Create request.
+type CreateOptsBuilder interface {
+	ToBGPVPNCreateMap() (map[string]any, error)
+}
+
+// CreateOpts represents options used to create a BGP VPN.
+type CreateOpts struct {
+	Name                string   `json:"name,omitempty"`
+	RouteDistinguishers []string `json:"route_distinguishers,omitempty"`
+	RouteTargets        []string `json:"route_targets,omitempty"`
+	ImportTargets       []string `json:"import_targets,omitempty"`
+	ExportTargets       []string `json:"export_targets,omitempty"`
+	LocalPref           int      `json:"local_pref,omitempty"`
+	VNI                 int      `json:"vni,omitempty"`
+	TenantID            string   `json:"tenant_id,omitempty"`
+	ProjectID           string   `json:"project_id,omitempty"`
+	Type                string   `json:"type,omitempty"`
+}
+
+// ToBGPVPNCreateMap builds a request body from CreateOpts.
+func (opts CreateOpts) ToBGPVPNCreateMap() (map[string]any, error) {
+	return gophercloud.BuildRequestBody(opts, "bgpvpn")
+}
+
+// Create a BGP VPN
+func Create(ctx context.Context, c *gophercloud.ServiceClient, opts CreateOpts) (r CreateResult) {
+	b, err := opts.ToBGPVPNCreateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	resp, err := c.Post(ctx, createURL(c), b, &r.Body, nil)
+	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
+	return
+}
+
+// Delete accepts a unique ID and deletes the BGP VPN associated with it.
+func Delete(ctx context.Context, c *gophercloud.ServiceClient, id string) (r DeleteResult) {
+	resp, err := c.Delete(ctx, deleteURL(c, id), nil)
+	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
+	return
+}
+
+// UpdateOptsBuilder allows extensions to add additional parameters to the
+// Update request.
+type UpdateOptsBuilder interface {
+	ToBGPVPNUpdateMap() (map[string]any, error)
+}
+
+// UpdateOpts represents options used to update a BGP VPN.
+type UpdateOpts struct {
+	Name                *string   `json:"name,omitempty"`
+	RouteDistinguishers *[]string `json:"route_distinguishers,omitempty"`
+	RouteTargets        *[]string `json:"route_targets,omitempty"`
+	ImportTargets       *[]string `json:"import_targets,omitempty"`
+	ExportTargets       *[]string `json:"export_targets,omitempty"`
+	LocalPref           *int      `json:"local_pref,omitempty"`
+}
+
+// ToBGPVPNUpdateMap builds a request body from UpdateOpts.
+func (opts UpdateOpts) ToBGPVPNUpdateMap() (map[string]any, error) {
+	return gophercloud.BuildRequestBody(opts, "bgpvpn")
+}
+
+// Update accept a BGP VPN ID and an UpdateOpts and update the BGP VPN
+func Update(ctx context.Context, c *gophercloud.ServiceClient, id string, opts UpdateOpts) (r UpdateResult) {
+	b, err := opts.ToBGPVPNUpdateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	resp, err := c.Put(ctx, updateURL(c, id), b, &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200},
```

Comment:
[Doc](https://docs.openstack.org/api-ref/network/v2/#update-a-bgp-vpn) says we're expecting 201 on success, can we double check this is correct?

Reply:
201 is not returned. You can confirm this in functional tests.